pkg/cmd/cli/cmd/expose.go

```diff
@@ -99,6 +107,7 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 			// Set default protocol back for generating services
 			if len(cmdutil.GetFlagString(cmd, "protocol")) == 0 {
 				cmd.Flags().Set("protocol", "TCP")
+				cmd.Flag("protocol").Changed = false
```

Comment:
Hey, sorry about not leaving a comment here. You may want to have a look at https://github.com/openshift/origin/pull/5968 for some context.